### PR TITLE
`pstr_prepare_scenarios()` now handles "weo" data correctly

### LINKED ISSUE #308: In PSTR with `type` "weo", each `value` is `NA`
``` r
library(dplyr, warn.conflicts = FALSE)
library(tiltIndicator)
packageVersion("tiltIndicator")
#> [1] '0.0.0.9056'

.type <- "weo"
companies <- pstr_companies |>
  filter(type == .type) |>
  filter(company_id %in% first(company_id))
scenarios <- pstr_scenarios |>
  filter(type == .type)

pstr(companies, scenarios) |> select(-companies_id)
#> # A tibble: 3 × 3
#>   grouped_by risk_category value
#>   <chr>      <chr>         <dbl>
#> 1 weo_NA_NA  high             NA
#> 2 weo_NA_NA  medium           NA
#> 3 weo_NA_NA  low              NA
```

<sup>Created on 2023-05-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by kalashsinghal**
Closes #308 
Relates to #307 
Relates to #310 

Each type of scenario -- "ipr" and "weo" -- have columns for "sector" and "subsector" but the name of those columns is different. That causes a join to not match what should be matched and result in `NA` in all rows of the "weo" type.

This PR fixes the problem by giving those columns the same name . The fix happens during preparation via `pstr_prepare_scenarios()`. WARNING: This is NOT main-stream code and the user would need to run `pstr_prepare_scenarios()` BEFORE they call `pstr()` and friends.

Because it's likely the user will forget, and the problem is silent, we'll follow up with another PR that detects the problem and throws an informative error message (see #311).

----

TODO

- [x] [Link related issue/PR]([url](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)). 
- [x] Describe the goal of the PR. Avoid details that are clear in the diff.
- [x] Mark the PR as draft.
- [x] [Include a unit test](https://code-review.tidyverse.org/reviewer/aspects.html#sec-tests).
- [x] Review your own PR in "Files changed".
- [x] Ensure the PR branch is updated.
- [x] Ensure the checks pass.
- [x] Change the status from draft to ready.
- [x] Polish the PR title and description.
- [x] Assign a reviewer.

EXCEPTIONS

- [ ] Slide here any item that you intentionally choose to not do.

### CODE REVIEW

**Comment by maurolepore** [comment]
Thanks @kalashsinghal, 

The reprex that before exposed the problem now seems to show correct results. 

However from "Files changed" I don't understand why you did what you did or how it fixes the problem. Can you explain it in words?

Also, for every bug we need a test that makes sure it never comes back. That test can be basically this reprex plus some `expect_*()` function.

``` r
library(dplyr, warn.conflicts = FALSE)
devtools::load_all()
#> ℹ Loading tiltIndicator
packageVersion("tiltIndicator")
#> [1] '0.0.0.9056'

.type <- "weo"
companies <- pstr_companies |>
  filter(type == .type) |>
  filter(company_id %in% first(company_id))
scenarios <- pstr_scenarios |>
  filter(type == .type)

pstr(companies, scenarios) |> 
  select(-companies_id) |> 
  print(n = Inf)
#> # A tibble: 36 × 3
#>    grouped_by                                   risk_category value
#>    <chr>                                        <chr>         <dbl>
#>  1 weo_announced pledges scenario_2020          high              0
#>  2 weo_announced pledges scenario_2020          medium            0
#>  3 weo_announced pledges scenario_2020          low               1
#>  4 weo_announced pledges scenario_2030          high              0
#>  5 weo_announced pledges scenario_2030          medium            0
#>  6 weo_announced pledges scenario_2030          low               1
#>  7 weo_announced pledges scenario_2040          high              0
#>  8 weo_announced pledges scenario_2040          medium            0
#>  9 weo_announced pledges scenario_2040          low               1
#> 10 weo_announced pledges scenario_2050          high              0
#> 11 weo_announced pledges scenario_2050          medium            0
#> 12 weo_announced pledges scenario_2050          low               1
#> 13 weo_net zero emissions by 2050 scenario_2020 high              0
#> 14 weo_net zero emissions by 2050 scenario_2020 medium            0
#> 15 weo_net zero emissions by 2050 scenario_2020 low               1
#> 16 weo_net zero emissions by 2050 scenario_2030 high              0
#> 17 weo_net zero emissions by 2050 scenario_2030 medium            0
#> 18 weo_net zero emissions by 2050 scenario_2030 low               1
#> 19 weo_net zero emissions by 2050 scenario_2040 high              0
#> 20 weo_net zero emissions by 2050 scenario_2040 medium            0
#> 21 weo_net zero emissions by 2050 scenario_2040 low               1
#> 22 weo_net zero emissions by 2050 scenario_2050 high              0
#> 23 weo_net zero emissions by 2050 scenario_2050 medium            0
#> 24 weo_net zero emissions by 2050 scenario_2050 low               1
#> 25 weo_stated policies scenario_2020            high              0
#> 26 weo_stated policies scenario_2020            medium            0
#> 27 weo_stated policies scenario_2020            low               1
#> 28 weo_stated policies scenario_2030            high              0
#> 29 weo_stated policies scenario_2030            medium            0
#> 30 weo_stated policies scenario_2030            low               1
#> 31 weo_stated policies scenario_2040            high              0
#> 32 weo_stated policies scenario_2040            medium            0
#> 33 weo_stated policies scenario_2040            low               1
#> 34 weo_stated policies scenario_2050            high              0
#> 35 weo_stated policies scenario_2050            medium            0
#> 36 weo_stated policies scenario_2050            low               1
```

<sup>Created on 2023-05-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

File: R/pstr_prepare_scenario.R [hunk 1]
```diff
@@ -19,6 +19,15 @@
...
+  if (hasName(scenarios, "weo")) {
+    scenarios$weo <- rename(
+      scenarios$weo,
+      weo_sector = "weo_product",
+      weo_subsector = "weo_flow"
+    )
+  }
+
   imap_dfr(scenarios, ~ pstr_prepare_scenario_impl(.x, .y))
 }
```
Comment (maurolepore) [inline R/pstr_prepare_scenario.R:32]: We now rename inside `pstr_prepare_scenario()`.

Comment (kalashsinghal) [reply R/pstr_prepare_scenario.R:32]: @maurolepore Thanks! It makes more sense to include it in `prepare` function.

**Comment by maurolepore** [comment]
For the record, this reprex shows how the exact same code that didn't work before now does work. What changed is the function `pstr_prepare_scenarios()` -- so to benefit from the fix you'll need to use its latest version.

cc' @Tilmon 

``` r
library(dplyr, warn.conflicts = FALSE)
library(readr, warn.conflicts = FALSE)
options(readr.show_col_types = FALSE)

# Before
library(tiltIndicator)
packageVersion("tiltIndicator")
#> [1] '0.0.0.9055'

companies <- pstr_companies
scenarios <- list(
  ipr = read_csv(extdata_path("pstr_ipr_2022.csv")),
  weo = read_csv(extdata_path("pstr_weo_2022.csv"))
) |>
  pstr_prepare_scenario()

pstr(companies, scenarios) |> 
  filter(is.na(value))
#> # A tibble: 42 × 4
#>    companies_id                             grouped_by risk_category value
#>    <chr>                                    <chr>      <chr>         <dbl>
#>  1 bst-procontrol-gmbh_00000005104947-001   weo_NA_NA  high             NA
#>  2 bst-procontrol-gmbh_00000005104947-001   weo_NA_NA  medium           NA
#>  3 bst-procontrol-gmbh_00000005104947-001   weo_NA_NA  low              NA
#>  4 ca-coity-trg-aua-gmbh_00000384-001       weo_NA_NA  high             NA
#>  5 ca-coity-trg-aua-gmbh_00000384-001       weo_NA_NA  medium           NA
#>  6 ca-coity-trg-aua-gmbh_00000384-001       weo_NA_NA  low              NA
#>  7 cheries-baqu_neu316541-00101             weo_NA_NA  high             NA
#>  8 cheries-baqu_neu316541-00101             weo_NA_NA  medium           NA
#>  9 cheries-baqu_neu316541-00101             weo_NA_NA  low              NA
#> 10 fleischerei-stiefsohn_00000005219477-001 weo_NA_NA  high             NA
#> # ℹ 32 more rows

# Now
devtools::load_all()
#> ℹ Loading tiltIndicator
packageVersion("tiltIndicator")
#> [1] '0.0.0.9057'

companies <- pstr_companies
scenarios <- list(
  ipr = read_csv(extdata_path("pstr_ipr_2022.csv")),
  weo = read_csv(extdata_path("pstr_weo_2022.csv"))
) |>
  pstr_prepare_scenario()

pstr(companies, scenarios) |> 
  filter(is.na(value))
#> # A tibble: 0 × 4
#> # ℹ 4 variables: companies_id <chr>, grouped_by <chr>, risk_category <chr>,
#> #   value <dbl>
```

<sup>Created on 2023-05-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Pull request merged**